### PR TITLE
Schema: introduce {host,service}_state#is_overdue

### LINKED ISSUE #38: Indicate overdue

### PR DESCRIPTION
**Pull request by Al2Klimov**
fixes #38

### CODE REVIEW

**Comment by lippserd** [comment]
Schema changes are fine.

File: configobject/statesync/overdue.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+redis.call('DEL', tempOverdue)
+
+return res
+
+`)
+
+//Counter on how many host/service is_overdue have synced since the last logOverdueSyncCounters()
+var overdueSyncCounters = struct {
+	host, service uint64
+}{}
```
Comment (N-o-X) [inline configobject/statesync/overdue.go]: You're never increasing those counters & please fix the comment

File: configobject/statesync/overdue.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+var updateHostOverdue = connection.DbIoSeconds.WithLabelValues("mysql", "update host_state by host_id")
+var updateServiceOverdue = connection.DbIoSeconds.WithLabelValues("mysql", "update service_state by service_id")
+
+//Start the sync goroutines for hosts and services
```
Comment (N-o-X) [inline configobject/statesync/overdue.go]: please fix the comment

File: configobject/statesync/overdue.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+func startOverdueSync(super *supervisor.Supervisor) {
+	go syncOverdue(super, "host", updateHostOverdue)
+	go syncOverdue(super, "service", updateServiceOverdue)
+	go logOverdueSyncCounters()
+}
+
+//Logs the amount of synced is_overdue every 20 seconds
```
Comment (N-o-X) [inline configobject/statesync/overdue.go]: please fix the comment

File: configobject/statesync/overdue.go [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+			log.WithFields(log.Fields{
+				"host":    host,
+				"service": service,
+				"period":  20 * time.Second,
+			}).Infof("Synced some host and service overdue indicators")
+		}
+	}
+}
+
+//Tries to sync is_overdue of given object type every second
```
Comment (N-o-X) [inline configobject/statesync/overdue.go]: please fix the comment

File: connection/redis.go [hunk 5]
```diff
@@ -339,6 +345,132 @@ func (rdbw *RDBWrapper) HGetAll(key string) *redis.StringStringMapCmd {
 	}
 }
 
+// Wrapper for connection handling
```
Comment (N-o-X) [inline connection/redis.go]: please fix the comment

File: connection/redis.go [hunk 6]
```diff
@@ -339,6 +345,132 @@ func (rdbw *RDBWrapper) HGetAll(key string) *redis.StringStringMapCmd {
...
+			if !rdbw.CheckConnection(false) {
+				continue
+			}
+		}
+
+		return cmd
+	}
+}
+
+// Wrapper for connection handling
```
Comment (N-o-X) [inline connection/redis.go]: please fix the comment

File: connection/redis.go [hunk 7]
```diff
@@ -339,6 +345,132 @@ func (rdbw *RDBWrapper) HGetAll(key string) *redis.StringStringMapCmd {
...
```
Comment (N-o-X) [inline connection/redis.go]: please fix the comment

File: connection/redis.go [hunk 8]
```diff
@@ -339,6 +345,132 @@ func (rdbw *RDBWrapper) HGetAll(key string) *redis.StringStringMapCmd {
...
```
Comment (N-o-X) [inline connection/redis.go]: please fix the comment

File: connection/redis.go [hunk 9]
```diff
@@ -339,6 +345,132 @@ func (rdbw *RDBWrapper) HGetAll(key string) *redis.StringStringMapCmd {
...
```
Comment (N-o-X) [inline connection/redis.go]: please fix the comment

File: connection/redis.go [hunk 10]
```diff
@@ -339,6 +345,132 @@ func (rdbw *RDBWrapper) HGetAll(key string) *redis.StringStringMapCmd {
...
```
Comment (N-o-X) [inline connection/redis.go]: please fix the comment

**Pull request merged**